waiter/src/waiter/util/client_tools.clj

```diff
@@ -563,38 +563,38 @@
       (log/info ks "=" value))
     value))
 
+(defn default-scheduler-settings
+  "Return the default 'leaf-level' scheduler's settings map"
+  [waiter-url]
+  (let [{:keys [kind] :as top-settings} (-> waiter-url waiter-settings :scheduler-config)
+        settings (get top-settings (keyword kind))]
+    (if-let [default-sub-scheduler (:default-scheduler settings)]
+      (get-in settings [:components (keyword default-sub-scheduler)])
+      settings)))
+
+(defn retrieve-default-scheduler-name
+  "Return the name of the default 'leaf-level' scheduler"
+  [waiter-url & {:keys [settings] :or {settings nil}}]
+  (let [settings (or settings (waiter-settings waiter-url))
+        kind (get-in settings [:scheduler-config :kind])
+        default-scheduler (get-in settings [:scheduler-config (keyword kind) :default-scheduler])]
+    (or default-scheduler kind)))
+
 (defn marathon-url
   "Returns the Marathon URL setting"
-  [waiter-url & {:keys [verbose] :or {verbose false}}]
-  (setting waiter-url [:scheduler-config :marathon :url] :verbose verbose))
-
-(defn k8s-log-bucket-url
-  "Returns the S3 bucket url for K8s service log backups"
-  [waiter-url & {:keys [verbose] :or {verbose false}}]
-  (setting waiter-url [:scheduler-config :kubernetes :log-bucket-url] :verbose verbose))
+  [waiter-url]
+  (-> (default-scheduler-settings waiter-url) :url))
```

Comment:
This doesn't work, it is retrieving the url from a scheduler, not necessarily marathon. It should retrieve the marathon scheduler settings and then lookup `url`.

Reply:
this is only called when the scheduler is marathon

Reply:
updated